### PR TITLE
Light node: dpos chain request

### PR DESCRIPTION
**Pull request by jwangdx**
### Description

* Implemented the les handling logic for `getHeaderAndValidatorsMsg` and `HeaderAndValidatorsMsg`

### CODE REVIEW

File: miner/worker.go [hunk 1]
```diff
@@ -500,7 +500,7 @@ func (w *worker) makeCurrent(parent *types.Block, header *types.Header) error {
 
 	// construct dpos context from parent's dpos context root
 	chainDB := stateDB.Database().TrieDB().DiskDB().(ethdb.Database)
-	dposContext, err := types.NewDposContextFromProto(chainDB, parent.Header().DposContext)
+	dposContext, err := types.NewDposContextFromProto(types.NewFullDposDatabase(chainDB), parent.Header().DposContext)
```
Comment (xzhangdx) [inline miner/worker.go]: maybe is better to rename the func of `types.NewDposContextFromProto ` as `types.NewDposContextFromRoot`

File: core/blockchain_insert.go [hunk 2]
```diff
@@ -19,6 +19,8 @@ package core
 import (
 	"time"
 
+	"github.com/DxChainNetwork/godx/consensus"
```
Comment (xzhangdx) [inline core/blockchain_insert.go]: format import packages: remove empty line bewteen  internal packages

File: core/headerchain.go [hunk 3]
```diff
@@ -26,6 +26,8 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/DxChainNetwork/godx/trie"
```
Comment (xzhangdx) [inline core/headerchain.go]: format import package: remove empty line bewteen internal packages

File: core/types/dpos_context.go [hunk 4]
```diff
@@ -21,69 +21,123 @@ import (
...
+// NewFullDposDatabase creates a new FullDposDatabase based on a diskdb
+func NewFullDposDatabase(diskdb ethdb.Database) *FullDposDatabase {
+	tdb := trie.NewDatabase(diskdb)
+	return &FullDposDatabase{tdb}
 }
 
-func NewMinedCntTrie(root common.Hash, db *trie.Database) (*trie.Trie, error) {
-	return trie.NewTrieWithPrefix(root, minedCntPrefix, db)
+// OpenEpochTrie opens the epochTrie
+func (db *FullDposDatabase) OpenEpochTrie(root common.Hash) (DposTrie, error) {
```
Comment (xzhangdx) [inline core/types/dpos_context.go]: `OpenEpochTrie ` and the next few open trie funcs are the same with input parameter and output type, that looks a little duplicative.

Maybe is just better to use the general func `db.openTrie ` and modify it to public.

File: miner/worker.go (hunk 1 above)
Comment (jwangdx) [reply miner/worker.go]: Agreed and fixed

File: core/blockchain_insert.go (hunk 2 above)
Comment (jwangdx) [reply core/blockchain_insert.go]: Fixed. Thx

File: core/headerchain.go (hunk 3 above)
Comment (jwangdx) [reply core/headerchain.go]: Fixed. Thx

File: core/types/dpos_context.go (hunk 4 above)
Comment (jwangdx) [reply core/types/dpos_context.go]: Well, they seems to be quite the same for FullDposDatabase. But they are different in light.dposOdrDatabase, who shares the same interface as the FullDposDatabase. If we want to change the function, we need to also change in `LesDposDatabase`, which is not quite applicable.

File: les/handler.go [hunk 5]
```diff
@@ -26,6 +26,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/DxChainNetwork/godx/consensus/dpos"
```
[the remark this answers is not in the document]
Comment (jwangdx) [reply les/handler.go]: Fixed. Thx.

**Comment by jwangdx** [comment]
Closed and will be open in another branch

**Pull request closed without merging**